packages/consensus/fcob/storage.go

```diff
@@ -316,6 +331,7 @@ func (m *MessageMetadata) SetMessageOpinionFormed(messageOpinionFormed bool) (mo
 	}
 
 	m.messageOpinionFormed = messageOpinionFormed
+	m.opinionFormedTime = clock.SyncedTime()
```

Comment:
Either write lock `opinionFormedTimeMutex` or change the getter to use the `messageOpinionFormedMutex`